File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result;
        if (root == nullptr) { return result; }

        queue<TreeNode*> q;
        int cnt = 0;

        q.push(root);

        while (!q.empty()) {
            TreeNode* rightSide = nullptr;
            cnt = q.size();

            for (int i = 0; i < cnt; ++i) {
                TreeNode* cur = q.front();
                q.pop();
                if (cur) {
                    rightSide = cur;
                    q.push(cur->left);
                    q.push(cur->right);
                }
            }
            if (rightSide) {
                result.push_back(rightSide->val);
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:        
        trav = []

        Li = collections.deque()
        Li.append(root)

        while Li:
            qLen = len(Li)
            vals = []
            for i in range(qLen):
                node = Li.popleft()
                if node:
                    vals.append(node.val)
                    Li.append(node.left)
                    Li.append(node.right)
            if vals:
                trav.append(vals)

        return trav    

        
    
        